TeachingRecordSystem/src/TeachingRecordSystem.Core/Jobs/EWCWalesImportJob.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            foreach (BlobHierarchyItem blobhierarchyItem in blobPage.Values)
+            {
+                // A hierarchical listing may return both virtual directories and blobs.
+                if (blobhierarchyItem.IsPrefix)
+                {
+                    if (includeSubfolders)
+                    {
+                        // Call recursively with the prefix to traverse the virtual directory.
+                        var subfolderFileNames = await GetFileNamesAsync(containerClient, blobhierarchyItem.Prefix, true, cancellationToken);
+                        if (subfolderFileNames != null)
+                        {
+                            fileNames.AddRange(subfolderFileNames);
+                        }
+                    }
+                }
+                else
+                {
+                    fileNames.Add(blobhierarchyItem.Blob.Name);
+                }
+            }
+        }
+
+        return fileNames.ToArray();
+    }
+
+    private async Task<string[]> GetFileNamesAsync(BlobContainerClient containerClient, string prefix, bool includeSubfolders, CancellationToken cancellationToken)
+    {
+        var fileNames = new List<string>();
+        var resultSegment = containerClient.GetBlobsByHierarchyAsync(prefix: prefix, delimiter: _pickupFolder, cancellationToken: cancellationToken).AsPages();
+
+        // Enumerate the blobs returned for each page.
+        await foreach (Azure.Page<BlobHierarchyItem> blobPage in resultSegment)
+        {
+            foreach (BlobHierarchyItem blobhierarchyItem in blobPage.Values)
+            {
+                // A hierarchical listing may return both virtual directories and blobs.
+                if (blobhierarchyItem.IsPrefix)
+                {
+                    if (includeSubfolders)
+                    {
+                        // Call recursively with the prefix to traverse the virtual directory.
+                        var subfolderFileNames = await GetFileNamesAsync(containerClient, blobhierarchyItem.Prefix, true, cancellationToken);
+                        if (subfolderFileNames != null)
+                        {
+                            fileNames.AddRange(subfolderFileNames);
+                        }
+                    }
+                }
+                else
+                {
+                    fileNames.Add(blobhierarchyItem.Blob.Name);
+                }
+            }
+        }
+
+        return fileNames.ToArray();
+    }
+
+    public async Task<MemoryStream> GetDownloadStreamAsync(string fileName)
+    {
+        BlobContainerClient containerClient = blobServiceClient.GetBlobContainerClient(_storageContainer);
+        BlobClient blobClient = containerClient.GetBlobClient($"{fileName}");
+        var downloadStream = new MemoryStream();
+        await blobClient.DownloadToAsync(downloadStream);
+        downloadStream.Position = 0;
+        return downloadStream;
```

Comment:
Saves creating an extra `Stream`:
```suggestion
        var streamingResult = await blobClient.DownloadStreamingAsync();
        return streamingResult.Value.Content;
```